lib/dotcom_web/controllers/schedule/timetable_controller.ex

```diff
@@ -148,6 +148,7 @@ defmodule DotcomWeb.ScheduleController.TimetableController do
       schedules ->
         schedules
         |> Enum.reject(&Schedules.Schedule.no_times?/1)
+        |> Enum.filter(&(&1.route.line_id == route.line_id))
```

Comment:
There was a bug that sometimes returned schedules for train's from another route (usually having overlapping stops with the requested route).
This matches on line_id instead of route ID so that we can match on shuttle routes that's serving this route.

Reply:
Depending on how long that list of schedules is, you could combine the `reject` and `filter` into a single private function. Otherwise, you're traversing the list twice.